doc/input.md

```diff
@@ -179,22 +188,34 @@ class DraggableComponent extends PositionComponent with Draggable {
 
   // update and render omitted
 
-  bool _isDragged = false;
+  Vector2 dragDeltaPosition;
+  bool get isDragging => dragDeltaPosition != null;
+
+  @override
+  bool onDragStarted(int pointerId, Vector2 startPosition) {
+    dragDeltaPosition = startPosition - position;
+    return false;
+  }
+
+  @override
+  bool onDragUpdated(int pointerId, DragUpdateDetails details) {
+    final localCoords = gameRef.convertGlobalToLocalCoordinate(
+      details.globalPosition.toVector2(),
+    );
```

Comment:
due to the way we bind this, anything coming from flutter is global. localPosition on the events is set to be == global. The reason and documentation is widely explained on the flutter docs in the code, so I don't think this is the end of the world. Exposing these methods from the game is a great thing to give users flexibility. I am considering monkey patching the local field since with this refactor we have full control, but if I do that will be a followup, this is already too big.